Input/Input.js

```diff
@@ -474,7 +474,7 @@ const InputBase = kind({
 			return calcAriaLabel('', null, placeholder);
 		},
 		buttonLabel: ({placeholder, type, value}) => {
-			return (isPasswordType(type) ? convertToPasswordFormat(value) : value) || placeholder;
+			return (isPasswordType(type) ? convertToPasswordFormat(value) : value) || typeof value === 'number' ? value.toString() : placeholder;
```

Comment:
To make this logic more readable, what about updating this code like the below pseudo code?

```
if (value's type is not one of String or Number) {
	return placeholder;
} else {
	return isPasswordType(type) ? convertToPasswordFormat(value) : value
}
```

Reply:
Also, I found the same issue of value 0 during review from the upper function `buttonAriaLabel`.

```
		buttonAriaLabel: ({placeholder, type, value}) => {
			if (value) { // This line is to check whether value is null or undefined due to a missing prop value. So, type checking looks better here.
				...
			}
			...
		},
```